src/main/kotlin/no/skatteetaten/aurora/boober/feature/HerkimerVaultFeature.kt

```diff
@@ -0,0 +1,192 @@
+package no.skatteetaten.aurora.boober.feature
+
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.fkorotkov.kubernetes.metadata
+import com.fkorotkov.kubernetes.newSecret
+import io.fabric8.kubernetes.api.model.Secret
+import no.skatteetaten.aurora.boober.model.AuroraConfigFieldHandler
+import no.skatteetaten.aurora.boober.model.AuroraContextCommand
+import no.skatteetaten.aurora.boober.model.AuroraDeploymentSpec
+import no.skatteetaten.aurora.boober.model.AuroraResource
+import no.skatteetaten.aurora.boober.model.addEnvVarsToMainContainers
+import no.skatteetaten.aurora.boober.model.findSubKeys
+import no.skatteetaten.aurora.boober.service.HerkimerService
+import no.skatteetaten.aurora.boober.service.ResourceHerkimer
+import no.skatteetaten.aurora.boober.service.ResourceKind
+import no.skatteetaten.aurora.boober.utils.addIfNotNull
+import no.skatteetaten.aurora.boober.utils.boolean
+import no.skatteetaten.aurora.boober.utils.createEnvVarRefs
+import no.skatteetaten.aurora.boober.utils.ensureEndsWith
+import no.skatteetaten.aurora.boober.utils.ensureStartWith
+import no.skatteetaten.aurora.boober.utils.jsonMapper
+import no.skatteetaten.aurora.boober.utils.oneOf
+import org.apache.commons.codec.binary.Base64
+import org.springframework.stereotype.Service
+
+private const val HERKIMER_RESOURCE_KEY = "resources"
+private const val FEATURE_FIELD = HERKIMER_RESOURCE_KEY
+private const val HERKIMER_REPONSE_KEY = "herkimerResponse"
+private const val HERKIMER_SECRETS_KEY = "secrets"
+
+@Service
+class HerkimerVaultFeature(
+    val herkimerService: HerkimerService
+) : Feature {
+    override fun handlers(header: AuroraDeploymentSpec, cmd: AuroraContextCommand): Set<AuroraConfigFieldHandler> {
+        return cmd.applicationFiles.findSubKeys(FEATURE_FIELD).flatMap { key ->
+            setOf(
+                AuroraConfigFieldHandler(
+                    "$FEATURE_FIELD/$key/enabled",
+                    defaultValue = true,
+                    validator = { it.boolean() }),
+                AuroraConfigFieldHandler("$FEATURE_FIELD/$key/prefix"),
+                AuroraConfigFieldHandler(
+                    "$FEATURE_FIELD/$key/serviceClass",
+                    validator = { node -> node.oneOf(ResourceKind.values().map { it.toString() }) }),
+                AuroraConfigFieldHandler(
+                    "$FEATURE_FIELD/$key/multiple",
+                    defaultValue = false,
+                    validator = { it.boolean() })
+            )
+        }.toSet()
+    }
+
+    override fun createContext(
```

Comment:
Denne måten å bruke context på er den jeg synes vi bør bruke overalt. Man henter ut det man trenger for eksterne integrasjoner 1 steg. Vi kan også skille på om det er en context som kun er brukt til validering med validationContext flagget.